src/ai/ai.service.ts

```
@@ -0,0 +1,173 @@
+    fen: string,
+    opts: EngineOptions = {},
+  ): Promise<BestMove> {
+    const { depth = 13, skillLevel = 14, movetimeMs } = opts;
+
+    return new Promise<BestMove>((resolve, reject) => {
+      const engine = this.createEngine();
+      let resolved = false;
+
+      const cleanUp = () => {
+        try {
+          // @ts-ignore - some builds support terminate/close/noop
```

ESLint

[error] Use "@ts-expect-error" instead of "@ts-ignore", as "@ts-ignore" will do nothing if the following line is error-free.

(@typescript-eslint/ban-ts-comment)